test/test_two_qubits.py

```diff
@@ -2,445 +2,71 @@
+    """Compare Hamilonians"""
+    model = get_two_qubit_chip.pmap.model
+    hdrift, hks = model.get_Hamiltonians()
     assert (hdrift.numpy() - test_data["hdrift"].numpy() < 1).any()
     for key in hks:
         almost_equal(hks[key], test_data["hks"][key])
 
 
-def test_propagation() -> None:
+def test_propagation(get_two_qubit_chip) -> None:
+    """Test that result of the propagation code does not change."""
+    exp = get_two_qubit_chip
+    pmap = exp.pmap
+    result = exp.propagation(pmap.model, pmap.generator, pmap.instructions["rx90p[0]"])
+    propagator = result["U"]
     almost_equal(propagator, test_data["propagator"])
 
 
-def test_init_point() -> None:
+def test_init_point(get_OC_optimizer) -> None:
     """Check that a previous best point can be loaded as an initial point."""
+    opt = get_OC_optimizer
     opt.load_best("test/best_point_open_loop.c3log")
 
 
 @pytest.mark.slow
 @pytest.mark.tensorflow
 @pytest.mark.optimizers
 @pytest.mark.integration
-def test_optim_tf_sgd() -> None:
+def test_optim(get_OC_optimizer) -> None:
     """
-    check if optimization result is below 1e-2
+    check if optimization result is below 1e-1
     """
-    opt.store_unitaries = True
+    opt = get_OC_optimizer
     opt.optimize_controls()
-    assert opt.current_best_goal < 0.01
-
-
-@pytest.mark.optimizers
-@pytest.mark.slow
-@pytest.mark.integration
-@pytest.mark.tensorflow
-def test_bad_tf_sgd() -> None:
-    bad_tf_opt = OptimalControl(
-        dir_path=logdir,
-        fid_func=fidelities.average_infid_set,
-        fid_subspace=["Q1", "Q2"],
-        pmap=pmap,
-        algorithm=algorithms.tf_sgd,
-        options={"maxfun": 2},
-        run_name="better_X90_bad_tf",
-    )
-    bad_tf_opt.set_exp(exp)
-
-    with pytest.raises(KeyError):
-        bad_tf_opt.optimize_controls()
-
-
-@pytest.mark.optimizers
-@pytest.mark.slow
-@pytest.mark.integration
-def test_optim_lbfgs() -> None:
-    lbfgs_opt = OptimalControl(
-        dir_path=logdir,
-        fid_func=fidelities.average_infid_set,
-        fid_subspace=["Q1", "Q2"],
-        pmap=pmap,
-        algorithm=algorithms.lbfgs,
-        options={"maxfun": 2},
-        run_name="better_X90_lbfgs",
-    )
-    lbfgs_opt.set_exp(exp)
-
-    lbfgs_opt.optimize_controls()
-    assert lbfgs_opt.current_best_goal < 0.01
-
-
-@pytest.mark.optimizers
-@pytest.mark.slow
-@pytest.mark.integration
-def test_optim_lbfgs_grad_free() -> None:
-    lbfgs_grad_free_opt = OptimalControl(
-        dir_path=logdir,
-        fid_func=fidelities.average_infid_set,
-        fid_subspace=["Q1", "Q2"],
-        pmap=pmap,
-        algorithm=algorithms.lbfgs_grad_free,
-        options={"maxfun": 5},
-        run_name="grad_free_lbfgs",
-    )
-    lbfgs_grad_free_opt.set_exp(exp)
-
-    lbfgs_grad_free_opt.optimize_controls()
-    assert lbfgs_grad_free_opt.current_best_goal < 0.01
+    assert opt.current_best_goal < 0.1
 
 
-def test_rk4() -> None:
+def test_rk4(get_two_qubit_chip) -> None:
```

Comment:
`pytest` markers, and maybe include an assert statement instead of just relying on this raising an error if RK4 is not working?